modules/gameSearch/src/main/GameSearchForm.scala

```diff
@@ -132,9 +132,23 @@ private[gameSearch] case class SearchSort(
     order: String = Sorting.default.order
 )
 
-private[gameSearch] case class SearchClock(
-    initMin: Option[Int] = None,
-    initMax: Option[Int] = None,
-    incMin: Option[Int] = None,
-    incMax: Option[Int] = None
+private[gameSearch] case class SearchClock private (
+    initMin: Option[Int],
+    initMax: Option[Int],
+    incMin: Option[Int],
+    incMax: Option[Int]
 )
+
+private[gameSearch] object SearchClock:
+  def empty = new SearchClock(None, None, None, None)
+  def apply(
+      initMin: Option[Int] = None,
+      initMax: Option[Int] = None,
+      incMin: Option[Int] = None,
+      incMax: Option[Int] = None
+  ): SearchClock =
+    inline def isValid =
+      (initMin, initMax).mapN(_ > _).getOrElse(true) || (incMin, incMax).mapN(_ > _).getOrElse(true)
+    if isValid
+    then empty
+    else SearchClock(initMin, initMax, incMin, incMax)
```

Comment:
isn't the logic inverted here?

Reply:
oh yes, that logic is wrong in so many level 🤦 that's why I always need tests

Reply:
this is funny, because the logic was correct, but the name was wrong. Fixed it anyway in https://github.com/lichess-org/lila/pull/15425/commits/730a764cfe86bbfefb3293f4b2198df01b3b5b47